judgels-frontends/raphael/src/routes/contests/contests/single/clarifications/ContestClarificationsPage/ContestClarificationsPage.tsx

```diff
@@ -165,6 +199,36 @@ class ContestClarificationsPage extends React.Component<
       this.setState({ isAnswerBoxLoading: false });
     }
   };
+
+  private renderFilterWidget = () => {
+    const { response, filter, isFilterLoading } = this.state;
+    if (!response) {
+      return null;
+    }
+    const { config } = response;
+    const { canSupervise } = config;
+    if (!canSupervise) {
+      return null;
+    }
+
+    const { status } = filter;
+    return (
+      <ClarificationFilterWidget
+        statuses={['ASKED']}
+        status={status}
+        onFilter={this.onFilter}
+        isLoading={!!isFilterLoading}
+      />
+    );
+  };
+
+  private onFilter = async filter => {
+    this.props.onAppendRoute(filter);
+    await this.setState({
```

Comment:
I suggest we always set the filter state based on query params, to make sure that they are always consistent.

Can we try similar approach to this? https://github.com/ia-toki/judgels/blob/master/judgels-frontends/raphael/src/routes/contests/contests/single/submissions/Programming/ContestSubmissionsPage/ContestSubmissionsPage.tsx#L63

Reply:
make sense, will update